compiler/rustc_interface/src/passes.rs

```diff
@@ -182,13 +182,15 @@ fn configure_and_expand(
                     new_path.push(path);
                 }
             }
-            env::set_var(
-                "PATH",
-                &env::join_paths(
-                    new_path.iter().filter(|p| env::join_paths(iter::once(p)).is_ok()),
-                )
-                .unwrap(),
-            );
+            unsafe {
+                env::set_var(
```

Comment:
This one is Windows-only where the underlying operation is *not* racy -- so that can probably be justified with a comment immediately.

Reply:
Makes me think if we should have `std::os::windows::set_var` which is safe 🤔